prediction_market_agent_tooling/tools/contract.py

```
@@ -654,6 +673,29 @@ def minimal_proxy_implements_function(
+        bytecode = web3.eth.get_code(contract_address)
+        logic_contract_address = bytecode[11:31]
+
+        return contract_implements_function(
+            Web3.to_checksum_address(logic_contract_address),
+            function_name=function_name,
+            web3=web3,
+            function_arg_types=function_arg_types,
+            look_for_proxy_contract=False,
+        )
+    except DecodingError:
+        logger.info(f"Error decoding contract address for singleton")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)